src/engine/PiecewiseLinearConstraint.h

```diff
@@ -254,6 +255,37 @@ class PiecewiseLinearConstraint : public ITableau::VariableWatcher
     */
     virtual void getCostFunctionComponent( Map<unsigned, double> &/* cost */ ) const {}
 
+    /*
+      Ask the piecewise linear constraint to add its cost term corresponding to
+      the given phase to the cost function.
+      Nothing should be added when the constraint is fixed or inactive.
+      Minimizing the added term should lead to the constraint being
+      "closer to satisfied" in the given phase status.
+    */
+    virtual void addCostFunctionComponent( Map<unsigned, double> &/* cost */,
+                                           PhaseStatus /* phase */ ) const {}
+
+    /*
+      Compute the error of the current assignment with respect to the cost compoenent
+      corresponding to the given phase.
+    */
+    virtual double computeCostFunctionComponent( PhaseStatus /* phase */ ) const
```

Comment:
Due to their names, it sounds like {add/compute}CostFunctionComponent are doing similar things. But I guess compute is computing not the component, but its value?

Reply:
Yeah exactly. add is adding a "symbolic term" to the cost function, compute returns the values of the term based on the current assignment.